### ISSUE TITLE
Remove dependency on jQuery

### ISSUE THREAD

**Issue by gregtyler** [ministryofjustice/moj-frontend#134]
## Summary

It would be nice to eliminate our dependency on jQuery, instead using native browser functionality (e.g. `setAttribute` instead of `attr`, `querySelector` rather than `find`).

## Motivation

There are a few reasons I think this is beneficial:
- It removes a dependency
- None of the jQuery functionality we use isn't achievable in a (fairly) modern browser
- It removes a barrier to entry for contributors: they wouldn't need to know jQuery
- The code is more aligned to current best practices, documentation and tutorials

## Describe alternatives you've considered

We can just keep using jQuery and accept the concerns above for now.

## Additional context

We'll need to add polyfills for functionality that isn't available in all browsers > IE8. (e.g. [Opera Mini and IE9/10 don't support `HTMLElement.dataset`](https://caniuse.com/mdn-api_htmlelement_dataset)) We may be able to detect and polyfill automatically as part of the build process.

We may also need to bulk out the IE8 polyfill if that's not already done by the GOV.UK Design system polyfill.

We shouldn't attempt this without unit tests (#132) as it would be difficult to spot regressions.

**Comment by DanCorder** [comment]
If this is going to take some time (it doesn't sound trivial) then in the meantime it would be nice to make the dependency on jQuery explicit. (I can't see it mentioned in the docs or the package.json)

**Comment by gregtyler** [comment]
Good point @DanCorder, we should add it to the [installing with NPM](https://github.com/ministryofjustice/moj-frontend/blob/main/docs/installation/installing-with-npm.md) documentation.

**Comment by gregtyler** [comment]
GDS are considering [removing jQuery from the prototype kit](https://github.com/alphagov/govuk-prototype-kit/issues/482), which would mean that (until we remove jQuery ourselves) we'd need to bundle jQuery in the package and add it to the [extension config file](https://github.com/ministryofjustice/moj-frontend/blob/fac61c1418d206280da40eefb3e982dff286a3e8/package/govuk-prototype-kit.config.json).

**Comment by gregtyler** [comment]
jQuery [is listed as a peer dependency](https://github.com/ministryofjustice/moj-frontend/blob/c0c700d44a99d1da1071850ca87433253cd18241/package/package.json#L33-L35) in `package.json` so, from what I understand of peer dependencies, your package manager should either auto-install it (this is what `npm` seems to do) or warn you to install it manually.

**Comment by yndajas** [comment]
> your package manager should either auto-install it (this is what `npm` seems to do) or warn you to install it manually.

As far as I can tell it didn't do that. `npm install` output after removing it as a direct dependency on our app but retaining the `@ministryofjustice/frontend` dependency:

```
up to date, audited 1177 packages in 785ms

191 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

**Comment by robertjmccarthy** [comment]
Closing this ticket as new ticket created #901
